diff --git a/src/tools/CodefulSdkGenerator/DirectClient/DirectClientTypeScriptCodeGenerator.cs b/src/tools/CodefulSdkGenerator/DirectClient/DirectClientTypeScriptCodeGenerator.cs
index 228d91aaff1..583a068af15 100644
--- a/src/tools/CodefulSdkGenerator/DirectClient/DirectClientTypeScriptCodeGenerator.cs
+++ b/src/tools/CodefulSdkGenerator/DirectClient/DirectClientTypeScriptCodeGenerator.cs
@@ -70,7 +70,7 @@ namespace Microsoft.Azure.Workflows.CodefulSdkGenerator.DirectClient
             stringBuilder.AppendLine();
             if (actionMethods.Count > 0)
             {
-                stringBuilder.AppendLine("import { AbortSignalLike } from \"@azure/abort-controller\";");
+                stringBuilder.AppendLine("import type { AbortSignalLike } from \"@azure/abort-controller\";");
             }
 
             stringBuilder.AppendLine("import { ConnectorClientBase } from \"../azureConnectors/clientBase.ts\";");
@@ -521,8 +521,8 @@ namespace Microsoft.Azure.Workflows.CodefulSdkGenerator.DirectClient
 
             if (method.HasResponse)
             {
-                stringBuilder.AppendLine($"        const url = this.resolveUrl(requestPath);");
-                stringBuilder.AppendLine($"        const httpResponse = await this.httpClient.sendAsync<{method.ResponseTypeName}>(\"{httpMethod}\", url{bodyArg});");
+                stringBuilder.AppendLine($"        const requestUrl = this.resolveUrl(requestPath);");
+                stringBuilder.AppendLine($"        const httpResponse = await this.httpClient.sendAsync<{method.ResponseTypeName}>(\"{httpMethod}\", requestUrl{bodyArg});");
                 stringBuilder.AppendLine();
                 stringBuilder.AppendLine("        if (!httpResponse.isSuccessStatusCode) {");
                 stringBuilder.AppendLine($"            throw new ConnectorException(this.connectorName, `{httpMethod} ${{requestPath}}`, httpResponse.statusCode, httpResponse.text);");
@@ -532,8 +532,8 @@ namespace Microsoft.Azure.Workflows.CodefulSdkGenerator.DirectClient
             }
             else
             {
-                stringBuilder.AppendLine($"        const url = this.resolveUrl(requestPath);");
-                stringBuilder.AppendLine($"        const httpResponse = await this.httpClient.sendAsync<void>(\"{httpMethod}\", url{bodyArg});");
+                stringBuilder.AppendLine($"        const requestUrl = this.resolveUrl(requestPath);");
+                stringBuilder.AppendLine($"        const httpResponse = await this.httpClient.sendAsync<void>(\"{httpMethod}\", requestUrl{bodyArg});");
                 stringBuilder.AppendLine();
                 stringBuilder.AppendLine("        if (!httpResponse.isSuccessStatusCode) {");
                 stringBuilder.AppendLine($"            throw new ConnectorException(this.connectorName, `{httpMethod} ${{requestPath}}`, httpResponse.statusCode, httpResponse.text);");
